src/Kinto.elm

```diff
@@ -472,6 +509,62 @@ extractKintoError statusCode statusMsg body =
                 |> ServerError statusCode statusMsg
 
 
+withQueryParam : ( String, String ) -> Request a -> Request a
+withQueryParam param builder =
+    let
+        ( url, params ) =
+            case String.split "?" builder.url of
```

Comment:
Have you tried using https://package.elm-lang.org/packages/elm/url/latest/Url-Parser ? It seems like it would be less brittle, don't you think?

Reply:
I am not sure of how to do that without creating a parser, I just want to split the string not defining the expected fields and values.

Reply:
Refs https://stackoverflow.com/questions/56234065/how-to-parse-a-querystring-in-elm-to-build-a-new-one-with-a-new-queryparameter